clients/js/src/embeddings/webai/WebAIEmbeddingFunction.ts

```diff
@@ -0,0 +1,244 @@
+import { BaseEmbeddingFunction } from "../IEmbeddingFunction";
+
+export type WebAIEmbeddingFunctionOptions = {
+  modality: "text" | "image" | "multimodal";
+  target: 'node' | 'browser';
+  proxy?: boolean;
+  wasmPath?: string;
+  modelID?: string;
+}
+
+export type WebAIEmbeddingMethods = { webAI: any, webAIText?: any, webAIImage?: any, webAIMultimodal?: any }
+
+/**
+ * `WebAIEmbeddingFunction` uses the Web AI package to generate embeddings.
+ * This embedding function can work in both NodeJS and browser environments.
+ * 
+ * @example
+ * ```typescript
+ * import { WebAIEmbeddingFunction } from "chromadb/webai";
+ * 
+ * const options = {
+ *   modality: "text",
+ *   target: "node",
+ * };
+ * 
+ * const webAI = require('@visheratin/web-ai-node');
+ * const webAIText = require('@visheratin/web-ai-node/text');
+ * const webAIInstance = new WebAIEmbeddingFunction(options, { webAI, webAIText });
+ * 
+ * const embeddings = await webAIInstance.generate(["Hello World", "Another Text"]);
+ * ```
+ * @example Let the embedding function load the webai libraries on runtime using .init():
+ * ```typescript
+ * import { WebAIEmbeddingFunction } from "chromadb/webai";
+ * 
+ * const options = {
+ *   modality: "text",
+ *   target: "node",
+ * };
+ * 
+ * const embeddingFunction = new WebAIEmbeddingFunction(options);
+ * await embeddingFunction.init()
+ * 
+ * const embeddings = await embeddingFunction.generate(["Hello World", "Another Text"]);
+ * ```
+ * 
+ * @remarks
+ * Browser version of Web AI (@visheratin/web-ai) is an ESM module.
+ * NodeJS version of Web AI (@visheratin/web-ai-node) is a CommonJS module.
+ */
+export class WebAIEmbeddingFunction extends BaseEmbeddingFunction<WebAIEmbeddingFunctionOptions, WebAIEmbeddingMethods> {
```

Comment:
@visheratin When you have a moment, could you please check and test whether the embedding function is functioning correctly?